libraries/esp8266/examples/MMU48K/MMU48K.ino

```diff
@@ -0,0 +1,354 @@
+                        "v_cfg = %u, state = %d, enable/disable count = %u/%u, "
+                        "map = 0x%02X, p = 0x%02X, v = 0x%02X}"),
+                   mmu_status.v_cfg, mmu_status.state,
+                   mmu_status.enable_count, mmu_status.disable_count,
+                   mmu_status.map, mmu_status.p, mmu_status.v);
+  oStream.println();
+  uint32_t iram_bank_reg = ESP8266_DREG(0x24);
+  if (0 == (iram_bank_reg & 0x10)) {  // if bit clear, is enabled
+    oStream.printf_P(PSTR("  IRAM block mapped to:    0x40108000"));
+    oStream.println();
+  }
+  if (0 == (iram_bank_reg & 0x08)) {
+    oStream.printf_P(PSTR("  IRAM block mapped to:    0x4010C000"));
+    oStream.println();
+  }
+#ifdef MMU_ICACHE_SIZE
+  oStream.printf_P(PSTR("  ICACHE Size:             %u"), MMU_ICACHE_SIZE);
+  oStream.println();
+#endif
+#ifdef MMU_IRAM_SIZE
+  oStream.printf_P(PSTR("  IRAM Size:               %u"), MMU_IRAM_SIZE);
+  oStream.println();
+  const uint32_t iram_free = MMU_IRAM_SIZE - (uint32_t)((uintptr_t)_text_end - 0x40100000UL);
+  oStream.printf_P(PSTR("  IRAM free:               %u"), iram_free);
+  oStream.println();
+#endif
+  oStream.printf_P(PSTR("  IRAM _text_end:          %p"), _text_end);
+  oStream.println();
+#ifdef MMU_SEC_HEAP
+  oStream.printf_P(PSTR("  Secondary Heap at:       %p"), MMU_SEC_HEAP);
+  oStream.println();
+  oStream.printf_P(PSTR("  Secondary Heap Size:     %u"), MMU_SEC_HEAP_SIZE);
+  oStream.println();
+#endif
+}
+
+
+void setup() {
+  WiFi.persistent(false);
+  WiFi.mode(WIFI_OFF);
+  // Serial.begin(74880);
+  Serial.begin(115200);
+  delay(20);
+  Serial.printf_P(PSTR("\n\nSetup ...\n"));
+
+  print_mmu_status(Serial);
+
+#if defined(MMU_IRAM_HEAP)
+  {
+    HeapSelectIram ephemeral;
+    // Serial.printf_P(PSTR("ESP.getFreeHeap(): %u\n"), ESP.getFreeHeap());
+    gobble_sz = ESP.getFreeHeap() - UMM_OVERHEAD_ADJUST; // - 4096;
+    gobble = (uint32_t *)malloc(gobble_sz);
+  }
+  Serial.printf_P(PSTR("\nmalloc() from IRAM Heap:\n"));
+  Serial.printf_P(PSTR("  gobble_sz: %u\n"), gobble_sz);
+  Serial.printf_P(PSTR("  gobble:    %p\n"), gobble);
+
+#elif defined(MMU_SEC_HEAP)
+  gobble = (uint32_t *)MMU_SEC_HEAP;
+  gobble_sz = MMU_SEC_HEAP_SIZE;
+#endif
+
+#if (MMU_IRAM_SIZE > 0x8000) || defined(MMU_IRAM_HEAP) || defined(MMU_SEC_HEAP)
+  if (isValid(gobble)) {
+    // Put something in our new memory
+    for (size_t i = 0; i < (gobble_sz / 4); i++) {
+      gobble[i] = (uint32_t)&gobble[i];
+    }
+
+    // Now is it there?
+    dump_mem32(gobble, 32);
+    // dump_mem32(&gobble[gobble_sz / 4 / 2], 32);
+    dump_mem32(&gobble[gobble_sz / 4 - 32], 32);
+  }
+#endif
+
+  // Lets peak over the edge
+  Serial.printf_P(PSTR("\nPeek over the edge of memory at 0x4010C000\n"));
+  dump_mem32((void *)(0x4010C000 - 16 * 4), 32);
+
+  probe_b = (char *)gobble;
+  probe_s = (short *)((uintptr_t)gobble);
+  unaligned_probe_s = (short *)((uintptr_t)gobble + 1);
+
+}
+
+
+extern uint32_t mmu_non32xfer_count;
```

Comment:
Rather than do this, it's better practice to implement an accessor function that returns the value.

Reply:
Unless you feel there is some value in this count, I'll remove it?

It was more to convince me my code was executing. Beyond the initial development use, I see no need to expose it.

Reply:
Sure, remove away. I was wondering about its use.

Reply:
Done.